src/deepsparse/server/config.py

```diff
@@ -12,152 +12,151 @@
-            if "models" in config_dict
-            else []
-        )
-        config = ServerConfig(**config_dict)
 
-    if len(config.models) == 0:
-        raise ValueError(
-            "There must be at least one model to serve in the configuration "
-            "for the deepsparse inference server"
-        )
 
-    return config
+class ServerConfig(BaseModel):
+    num_cores: int = Field(
+        description="The number of cores available for model execution. "
+        "Defaults to all available cores.",
+    )
+
+    num_workers: int = Field(
+        description="The number of workers to split the available cores between."
+    )
+
+    endpoints: List[EndpointConfig] = Field(description="The models to serve.")
 
 
-def server_config_to_env(
-    config_file: str,
-    task: str,
-    model_path: str,
-    batch_size: int,
-    integration: str,
-    env_key: str = ENV_DEEPSPARSE_SERVER_CONFIG,
-):
+def _unpack_bucketing(
+    task: str, bucketing: Optional[Union[SequenceLengthsConfig, ImageSizesConfig]]
+) -> Tuple[Optional[List[int]], Dict[str, Any]]:
     """
-    Put a server configuration in an environment variable given by env_key.
-    If config_file is given, ignores task, model_path, and batch_size.
-    Otherwise, creates a configuration file from task, model_path, and batch_size
-    for serving a single model.
-
-    :param config_file: the path to the config file to store in the environment
-    :param task: the task the model_path is serving such as question_answering.
-        If config_file is supplied, this is ignored.
-    :param model_path: the path to a model.onnx file, a model folder containing
-        the model.onnx and supporting files, or a SparseZoo model stub.
-        If config_file is supplied, this is ignored.
-    :param batch_size: the batch size to serve the model from model_path with.
-        If config_file is supplied, this is ignored.
-    :param integration: name of deployment integration that this server will be
-        deployed to. Supported options include None for default inference and
-        sagemaker for inference deployment on AWS Sagemaker
-    :param env_key: the environment variable to set the configuration in.
-        Defaults to ENV_DEEPSPARSE_SERVER_CONFIG
+    :return: (input_shapes, kwargs) which are passed to PipelineConfig
     """
-    if config_file is not None:
-        config = config_file
-    else:
-        if task is None or model_path is None:
+    if bucketing is None:
+        return None, {}
+
+    if isinstance(bucketing, SequenceLengthsConfig):
+        if not SupportedTasks.is_nlp(task):
+            raise ValueError(f"SequenceLengthConfig specified for non-nlp task {task}")
+
+        return _unpack_nlp_bucketing(bucketing)
+    elif isinstance(bucketing, ImageSizesConfig):
+        if (
+            not SupportedTasks.is_image_classification(task)
+            and not SupportedTasks.is_yolo(task)
+            and not SupportedTasks.is_yolact(task)
```

Comment:
let's add an `is_cv` helper, probably won't be the last time we use this if statement / add another cv pipeline

Reply:
Can we do this in another PR? don't want to add an additional file to this PR to review

Reply:
it should be a minimal change with basically this same code no?  Just worried about this falling through the cracks. open an asana ticket if you feel strongly

Reply:
Added